src/lib/es5.d.ts

```diff
@@ -1446,9 +1446,7 @@ type Extract<T, U> = T extends U ? T : never;
 /**
  * Construct a type with the properties of T except for those in type K.
  */
-type Omit<T, K extends keyof any> = {

```

Comment:
🤔Why can't this be `K extends keyof T` to make the type homomorphic?

Reply:
I don't believe that would make the type homomorphic, would it @ahejlsberg?

Reply:
Yeah, discussed offline, but for reasons outlined below, the fact that it's not just a simple `keyof T` or `K` (where `K extends keyof T`) means that TypeScript can't trivially look at that and say "yup, we gotta keep the modifiers around on this one".

Reply:
Could also do
```ts
type Omit<T, K extends keyof any> = T extends unknown ? Pick<T, Exclude<keyof T, K>> : never;
```
that is to say: making the builtin `Omit` distributive. (Which'd give it an alias symbol, too) A bunch of library authors end up swapping their internal `Omit` to this once they get into situations involving unions.

Distributive conditional like this have perf issues when nested repeatedly right now, tho, sooooo.... ehhhhh?